src/components/StackNavigation/StackNavigation.tsx

```diff
@@ -0,0 +1,83 @@
+import {FC, useEffect} from "react";
+import {useNavigate} from "react-router";
+import {Note} from "types/note";
+import {ReactComponent as RightArrowIcon} from "assets/icon-arrow-next.svg";
+import {ReactComponent as LeftArrowIcon} from "assets/icon-arrow-previous.svg";
+import "./StackNavigation.scss";
+import {StackNavigationDots} from "./Dots/StackNavigationDots";
+
+interface StackNavigationProps {
+  stacks: Note[];
+  currentStack: string;
+  prevColumnStack: string | undefined;
+  nextColumnStack: string | undefined;
+  handleModeration: (stackId: string) => void;
+}
+
+export const StackNavigation: FC<StackNavigationProps> = ({stacks, currentStack, prevColumnStack, nextColumnStack, handleModeration}: StackNavigationProps) => {
+  const navigate = useNavigate();
+  const currentIndex = stacks.findIndex((s) => s.id === currentStack);
+
+  const handleBackClick = () => {
+    if (currentIndex > 0) {
+      handleModeration(stacks[currentIndex - 1].id);
+      navigate(`../note/${stacks[currentIndex - 1].id}/stack`);
+    } else if (prevColumnStack) {
+      handleModeration(prevColumnStack);
+      navigate(`../note/${prevColumnStack}/stack`);
+    }
+  };
+
+  const handleForwardClick = () => {
+    if (currentIndex < stacks.length - 1) {
+      handleModeration(stacks[currentIndex + 1].id);
+      navigate(`../note/${stacks[currentIndex + 1].id}/stack`);
+    } else if (nextColumnStack) {
+      handleModeration(nextColumnStack);
+      navigate(`../note/${nextColumnStack}/stack`);
+    }
+  };
+
+  const handleKeyDown = (event: KeyboardEvent) => {
+    if (event.key === "ArrowLeft" || event.key === "ArrowUp") {
+      event.preventDefault();
+      handleBackClick();
+    } else if (event.key === "ArrowRight" || event.key === "ArrowDown") {
+      event.preventDefault();
+      handleForwardClick();
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener("keydown", handleKeyDown);
+
+    return () => window.removeEventListener("keydown", handleKeyDown);
+  });
```

Comment:
Do you know a better way to do this? I first tried using our Hotkey functionality, but I couldn't get it to work because the functions here are dynamic (as in, we need to check the current ID, next stacks etc. on every navigate)